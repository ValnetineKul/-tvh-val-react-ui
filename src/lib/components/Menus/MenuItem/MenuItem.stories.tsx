import type { ComponentProps } from 'react';
import React from 'react';
import { action } from '@storybook/addon-actions';
import { useArgs } from '@storybook/client-api';
import { useSurface } from '../../../../themes/core';
import type { StoryTemplate } from '../../../story-utils';
import { createMeta } from '../../../story-utils';
import { AngleRight, FileDownload } from '../../Icon/icons/functional';
import Icon from '../../Icon';
import InlineMessage from '../../InlineMessage';
import Tag from '../../Tags/Tag';
import MenuItem from './MenuItem';

type Props = ComponentProps<typeof MenuItem>;

const Template = (args: Props) => {
  const {
    onCheckboxChange,
    checkboxProps,
    checkbox: hasCheckbox,
    indeterminate: isIndeterminate,
    ...remainingProps
  } = args;
  const { color: onSurface } = useSurface();
  return <MenuItem {...remainingProps} color={onSurface} onClick={action('onClick')} />;
};

const TemplateCheckbox = ((args: Props) => {
  const [{ selected }, updateArgs] = useArgs();
  const { color: onSurface } = useSurface();
  const handleChange = () => {
    updateArgs({ selected: !selected });
    action('onCheckboxChange')(selected);
  };
  const { onClick, buttonProps, startIcon, endIcon, subLabel, inlineMessage, ...remainingProps } = args;

  return (
    <MenuItem {...remainingProps} color={onSurface} checkbox selected={selected} onCheckboxChange={handleChange} />
  );
}) as StoryTemplate<Props>;

export const Primitive = Template.bind({});
Primitive.args = {
  label: 'Item Label',
  selected: false,
  disabled: false,
};
Primitive.storyName = 'label';

export const ComponentLabel = Template.bind({});
ComponentLabel.args = {
  label: <div>Component Label</div>,
  selected: false,
  disabled: false,
};
ComponentLabel.argTypes = {
  label: {
    control: {
      disable: true,
    },
  },
};
ComponentLabel.storyName = 'componentLabel';

export const StartIcon = Template.bind({});
StartIcon.args = {
  ...Primitive.args,
  startIcon: <Icon icon={FileDownload} />,
};
StartIcon.argTypes = {
  startIcon: {
    table: {
      disable: false,
    },
    control: {
      disable: true,
    },
  },
};
StartIcon.storyName = 'startIcon';

export const EndIcon = Template.bind({});
EndIcon.args = {
  ...Primitive.args,
  endIcon: <Icon icon={AngleRight} />,
};
EndIcon.argTypes = {
  endIcon: {
    table: {
      disable: false,
    },
    control: {
      disable: true,
    },
  },
};
EndIcon.storyName = 'endIcon';

export const StartAndEndIcon = Template.bind({});
StartAndEndIcon.args = {
  ...Primitive.args,
  startIcon: <Icon icon={FileDownload} />,
  endIcon: <Icon icon={AngleRight} />,
};
StartAndEndIcon.argTypes = {
  startIcon: {
    table: {
      disable: false,
    },
    control: {
      disable: true,
    },
  },
  endIcon: {
    table: {
      disable: false,
    },
    control: {
      disable: true,
    },
  },
};
StartAndEndIcon.storyName = 'startAndEndIcon';

export const Tags = Template.bind({});
Tags.args = {
  ...Primitive.args,
  tag: <Tag label="123" />,
};
Tags.argTypes = {
  tag: {
    table: {
      disable: false,
    },
    control: {
      disable: true,
    },
  },
};
Tags.storyName = 'tag';

export const InlineMessages = Template.bind({});
InlineMessages.args = {
  ...Primitive.args,
  inlineMessage: <InlineMessage message="Success message" status="success" />,
};
InlineMessages.argTypes = {
  inlineMessage: {
    table: {
      disable: false,
    },
    control: {
      disable: true,
    },
  },
};
InlineMessages.storyName = 'inlineMessage';

export const SubLabel = Template.bind({});
SubLabel.args = {
  ...Primitive.args,
  subLabel: 'Extra info',
};
SubLabel.argTypes = {
  subLabel: {
    table: {
      disable: false,
    },
  },
};
SubLabel.storyName = 'subLabel';

export const StartIconWithSubLabel = Template.bind({});
StartIconWithSubLabel.args = {
  ...Primitive.args,
  startIcon: <Icon icon={FileDownload} />,
  subLabel: 'Extra info',
};
StartIconWithSubLabel.argTypes = {
  subLabel: {
    table: {
      disable: false,
    },
  },
  startIcon: {
    table: {
      disable: false,
    },
    control: {
      disable: true,
    },
  },
};
StartIconWithSubLabel.storyName = 'startIconWithSubLabel';

export const StartIconWithSubLabelWithTag = Template.bind({});
StartIconWithSubLabelWithTag.args = {
  ...Primitive.args,
  tag: <Tag label="123" />,
  startIcon: <Icon icon={FileDownload} />,
  subLabel: 'Extra info',
};
StartIconWithSubLabelWithTag.argTypes = {
  tag: {
    table: {
      disable: false,
    },
    control: {
      disable: true,
    },
  },
};
StartIconWithSubLabelWithTag.storyName = 'startIconWithSubLabelWithTag';

export const Checkbox = TemplateCheckbox.bind({});
Checkbox.args = {
  ...Primitive.args,
  onCheckboxChange: () => {},
  checkbox: true,
};

Checkbox.argTypes = {
  checkbox: {
    table: {
      disable: false,
    },
    control: {
      disable: true,
    },
  },
  indeterminate: {
    table: {
      disable: false,
    },
  },
  onCheckboxChange: {
    table: {
      disable: false,
    },
  },
  onClick: {
    table: {
      disable: true,
    },
  },
  download: {
    table: {
      disable: true,
    },
  },
  to: {
    table: {
      disable: true,
    },
  },
  href: {
    table: {
      disable: true,
    },
  },
  target: {
    table: {
      disable: true,
    },
  },
  component: {
    table: {
      disable: true,
    },
  },
};

Checkbox.storyName = 'checkbox';

export const CheckboxWithTag = TemplateCheckbox.bind({});
CheckboxWithTag.args = {
  ...Primitive.args,
  onCheckboxChange: () => {},
  checkbox: true,
  tag: <Tag label="123" />,
};

CheckboxWithTag.argTypes = {
  checkbox: {
    table: {
      disable: false,
    },
    control: {
      disable: true,
    },
  },
  tag: {
    table: {
      disable: false,
    },
    control: {
      disable: true,
    },
  },
  indeterminate: {
    table: {
      disable: false,
    },
  },
  onCheckboxChange: {
    table: {
      disable: false,
    },
  },
  onClick: {
    table: {
      disable: true,
    },
  },
  download: {
    table: {
      disable: true,
    },
  },
  to: {
    table: {
      disable: true,
    },
  },
  href: {
    table: {
      disable: true,
    },
  },
  target: {
    table: {
      disable: true,
    },
  },
  component: {
    table: {
      disable: true,
    },
  },
};
CheckboxWithTag.storyName = 'checkboxWithTag';

export default createMeta({
  component: MenuItem,
  title: 'Components/Menus/MenuItem',
  argTypes: {
    checkbox: {
      table: {
        disable: true,
      },
    },
    subLabel: {
      table: {
        disable: true,
      },
    },
    startIcon: {
      table: {
        disable: true,
      },
    },
    endIcon: {
      table: {
        disable: true,
      },
    },
    tag: {
      table: {
        disable: true,
      },
    },
    inlineMessage: {
      table: {
        disable: true,
      },
    },
    subMenu: {
      table: {
        disable: true,
      },
    },
    indeterminate: {
      table: {
        disable: true,
      },
    },
    elementType: {
      control: {
        disable: true,
      },
      table: {
        type: { detail: 'React or HTML element type' },
      },
    },
    component: {
      control: {
        disable: true,
      },
      table: {
        type: { detail: 'React.ReactElement' },
      },
    },
    href: {
      control: {
        disable: true,
      },
      table: {
        type: { detail: 'If "href" is filled, the component will immediately transform to HTML link tag "a"' },
      },
    },
    to: {
      control: {
        disable: true,
      },
    },
    download: {
      control: {
        disable: true,
      },
      table: {
        type: {
          detail:
            'If "href" and "download" are filled â€“ downloads the file when clicking on link (instead of navigating to the file)',
        },
      },
    },
    index: {
      control: { disable: true },
    },
    focusVisibleClassName: {
      control: { disable: true },
    },
    labelClassName: {
      control: { disable: true },
    },
    onClick: {
      action: { argTypesRegex: '^on.*' },
      control: { disable: true },
      table: {
        type: {
          detail: 'Callback fired when MenuItem button/link is cklicked',
        },
      },
    },
    onCheckboxChange: {
      table: {
        disable: true,
        type: {
          detail: 'Callback fired when the checkbox state is changed',
        },
      },
    },
  },
});
