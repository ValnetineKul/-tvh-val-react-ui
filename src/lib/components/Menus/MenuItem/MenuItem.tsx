import React, { forwardRef, isValidElement } from 'react';
import { default as MuiMenuItem } from '@mui/material/MenuItem';
import TruncatedText from '../../TruncatedText/TruncatedText';
import ButtonBase from '../../Surfaces/ButtonBase';
import Checkbox from '../../Checkbox';
import useStyles from './MenuItem.styles';
import { useButtonBase } from '../../ButtonBase';
import type { MenuItemProps } from './MenuItem.types';

const MenuItem = forwardRef<HTMLLIElement, MenuItemProps>(
  (
    {
      label,
      selected = false,
      disabled = false,
      elementType = 'li',
      labelClassName,
      className,
      focusVisibleClassName,
      buttonBaseClassName,
      color = '100',
      style,
      tag,
      subLabel,
      onClick,
      onCheckboxChange,
      checkboxProps,
      checkbox: hasCheckbox,
      indeterminate: isIndeterminate,
      ...props
    },
    ref
  ) => {
    const { classes, cx } = useStyles({ surface: color });

    const startIcon = 'startIcon' in props && props.startIcon;
    const endIcon = 'endIcon' in props && props.endIcon;
    const inlineMessage = 'inlineMessage' in props && props.inlineMessage;
    const buttonProps = 'buttonProps' in props && props.buttonProps;

    const renderLabel = () => {
      return typeof label === 'string' ? (
        <TruncatedText
          text={label}
          typographyProps={{
            component: 'span',
            weight: selected ? ('emphasis' as const) : ('regular' as const),
            className: labelClassName,
          }}
          tooltipProps={{ className: classes.label }}
        />
      ) : (
        label
      );
    };

    const renderTag = () => {
      return React.isValidElement(tag)
        ? React.cloneElement(tag, { size: 'sm', className: cx(tag.props.className, classes.tag) })
        : undefined;
    };

    const renderCheckboxLabel = () => {
      if (tag) {
        return (
          <span className={classes.checkboxWithTagWrapper}>
            {renderLabel()}
            {renderTag()}
          </span>
        );
      }
      return renderLabel();
    };

    const buttonBaseProps = useButtonBase(props);

    return (
      <MuiMenuItem
        ref={ref}
        classes={{ root: classes.root, selected: classes.selected, focusVisible: classes.focusVisible }}
        className={cx({ [classes.disabled]: disabled }, className)}
        selected={selected}
        disabled={disabled}
        component={elementType}
        style={style}
      >
        {hasCheckbox ? (
          <>
            {onCheckboxChange && (
              <Checkbox
                inputProps={checkboxProps}
                onChange={onCheckboxChange}
                className={classes.label}
                formControlLabelClassName={focusVisibleClassName}
                label={renderCheckboxLabel()}
                indeterminate={isIndeterminate}
                checked={selected}
                disabled={disabled}
              />
            )}
          </>
        ) : (
          <ButtonBase
            {...buttonProps}
            {...buttonBaseProps}
            color={color}
            onClick={onClick}
            className={cx(classes.wrapper, focusVisibleClassName, buttonBaseClassName)}
            focus="inset"
            disabled={disabled}
          >
            {startIcon && isValidElement(startIcon)
              ? React.cloneElement(startIcon, { size: 'md', className: classes.startIcon })
              : startIcon}
            <div className={cx(classes.labelWrapper, classes.label)}>
              {renderLabel()}
              {subLabel &&
                (typeof subLabel === 'string' ? (
                  <TruncatedText
                    text={subLabel}
                    typographyProps={{
                      component: 'div',
                      variant: 'body400',
                      secondary: true,
                      className: classes.subLabel,
                    }}
                  />
                ) : (
                  subLabel
                ))}
            </div>
            {inlineMessage &&
              React.isValidElement(inlineMessage) &&
              React.cloneElement(inlineMessage, {
                size: 'sm',
                className: cx(inlineMessage.props.className, classes.inlineMessage),
              })}
            {endIcon &&
              React.isValidElement(endIcon) &&
              React.cloneElement(endIcon, { size: 'md', className: cx(endIcon.props.className, classes.endIcon) })}
            {tag && renderTag()}
          </ButtonBase>
        )}
      </MuiMenuItem>
    );
  }
);
MenuItem.displayName = 'MenuItem';

export default React.memo(MenuItem);
