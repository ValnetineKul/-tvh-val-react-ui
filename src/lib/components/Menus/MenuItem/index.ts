import MenuItem from './MenuItem';

export type { MenuItemProps } from './MenuItem.types';
export default MenuItem;
