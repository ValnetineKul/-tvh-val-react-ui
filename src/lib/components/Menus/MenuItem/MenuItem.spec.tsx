import type { ComponentProps } from 'react';
import React from 'react';
import { render, userEvent, screen } from '../../../test-utils';
import MenuItem from './MenuItem';
import { AngleRight, FileDownload } from '../../Icon/icons/functional';
import Icon from '../../Icon';

import {
  Primitive,
  StartIcon,
  EndIcon,
  Tags,
  InlineMessages,
  SubLabel,
  Checkbox,
  ComponentLabel,
  CheckboxWithTag,
  StartIconWithSubLabel,
} from './MenuItem.stories';
import type { Story } from '../../../story-utils';

const props = {
  startIcon: <Icon icon={FileDownload} />,
  endIcon: <Icon icon={AngleRight} />,
  onClick: jest.fn(),
};

const checkboxProps = {
  checkbox: true,
  onCheckboxChange: jest.fn(),
};

describe('MenuItem', () => {
  beforeAll(() => {
    jest.useFakeTimers();
    jest.clearAllMocks();
  });
  afterAll(() => {
    jest.useRealTimers();
  });

  describe('Should render stories correctly', () => {
    const cases: [string, Story<ComponentProps<typeof MenuItem>>][] = [
      ['text', Primitive],
      ['component', ComponentLabel],
      ['start icon', StartIcon],
      ['end icon', EndIcon],
      ['tag', Tags],
      ['inline message', InlineMessages],
      ['sub label', SubLabel],
      ['start icon with sub label', StartIconWithSubLabel],
      ['checkbox', Checkbox],
      ['checkbox with tag', CheckboxWithTag],
    ];
    test.each(cases)('story: %p', (testName, storybookCase) => {
      const { container } = render(<MenuItem {...storybookCase.args} />);
      expect(container.firstChild).toMatchSnapshot(testName);
    });
  });

  it('Should ignore invalid tag', () => {
    render(<MenuItem label="example label" tag="invalid tag" />);
    expect(screen.queryByText('invalid tag')).not.toBeInTheDocument();
  });

  it('Should show tooltip on hover', async () => {
    const { rerender, container } = render(<MenuItem {...props} label="Long label" />);
    expect(container.querySelector('[role="tooltip"]')).toBeNull();

    const label = screen.getByText('Long label');

    Object.defineProperties(label, {
      scrollWidth: { configurable: true, value: 144 },
      offsetWidth: { configurable: true, value: 40 },
    });

    rerender(<MenuItem {...props} label="Long long label" />);

    const label1 = screen.getByText('Long long label');
    userEvent.hover(label1);

    const tooltip = await screen.findByRole('tooltip');
    expect(tooltip).toBeInTheDocument();
  });

  it('Should trigger onClick', () => {
    const onClick = jest.fn();
    render(<MenuItem {...props} onClick={onClick} label="Label" />);
    const menuItemButton = screen.getByRole('button');
    menuItemButton.click();
    expect(onClick).toHaveBeenCalledTimes(1);
  });

  it('Should trigger onCheckboxChange', () => {
    const onCheckboxChange = jest.fn();
    render(<MenuItem {...checkboxProps} onCheckboxChange={onCheckboxChange} label="Label" />);
    const checkbox = screen.getByRole('checkbox') as HTMLInputElement;
    userEvent.click(checkbox);
    expect(onCheckboxChange).toHaveBeenCalledTimes(1);
  });
});
