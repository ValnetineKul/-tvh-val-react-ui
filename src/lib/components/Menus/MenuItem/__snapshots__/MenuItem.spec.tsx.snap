// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuItem Should render stories correctly story: "checkbox with tag": checkbox with tag 1`] = `
<li
  class="tss-1qfvypo-MenuItem-root MuiMenuItem-root MuiMenuItem-gutters MuiButtonBase-root  mui-t7a2rv-MuiButtonBase-root-MuiMenuItem-root"
  role="menuitem"
  tabindex="-1"
>
  <label
    class="tss-n075z2-Checkbox-root MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd mui-j204z7-MuiFormControlLabel-root"
  >
    <span
      class="MuiCheckbox-root MuiCheckbox-colorPrimary MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root tss-hpetsa-Checkbox-checkbox mui-355xbt-MuiButtonBase-root-MuiCheckbox-root"
    >
      <input
        class="PrivateSwitchBase-input mui-1m9pwf3"
        data-indeterminate="false"
        type="checkbox"
      />
      <span
        class="tss-1vu0oqh-Checkbox-icon-Checkbox-unchecked"
        font-size="medium"
      />
    </span>
    <span
      class="MuiTypography-root MuiTypography-body1 tss-5-Checkbox-label-ref tss-fxezff-Checkbox-label-MenuItem-label MuiFormControlLabel-label mui-b600p5-MuiTypography-root"
    >
      <span
        class="tss-0-Typography-weightRegular-ref tss-6tt2d9-Typography-root-Typography-weightRegular-Typography-body500"
      >
        <span
          class="tss-1030q1d-MenuItem-checkboxWithTagWrapper"
        >
          <div
            class="tss-0-Typography-weightRegular-ref tss-10hdle-Typography-root-Typography-weightRegular-Tooltip-wrapper-TruncatedText-truncatedText-MenuItem-label-Typography-body400"
            data-mui-internal-clone-element="true"
          >
            <span
              class="tss-0-Typography-weightRegular-ref tss-1hhfpop-Typography-root-Typography-weightRegular-TruncatedText-truncatedText-Typography-body500"
            >
              Item Label
            </span>
          </div>
          <div
            class="tss-6-MenuItem-tag-ref tss-4u6r2-Tag-root-Tag-colorPrimary-MenuItem-tag"
          >
            <span
              class="tss-0-Typography-weightRegular-ref tss-j8p12z-Typography-root-Typography-weightRegular-Typography-body400"
            >
              123
            </span>
          </div>
        </span>
      </span>
    </span>
  </label>
</li>
`;

exports[`MenuItem Should render stories correctly story: "checkbox": checkbox 1`] = `
<li
  class="tss-1qfvypo-MenuItem-root MuiMenuItem-root MuiMenuItem-gutters MuiButtonBase-root  mui-t7a2rv-MuiButtonBase-root-MuiMenuItem-root"
  role="menuitem"
  tabindex="-1"
>
  <label
    class="tss-n075z2-Checkbox-root MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd mui-j204z7-MuiFormControlLabel-root"
  >
    <span
      class="MuiCheckbox-root MuiCheckbox-colorPrimary MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root tss-hpetsa-Checkbox-checkbox mui-355xbt-MuiButtonBase-root-MuiCheckbox-root"
    >
      <input
        class="PrivateSwitchBase-input mui-1m9pwf3"
        data-indeterminate="false"
        type="checkbox"
      />
      <span
        class="tss-1vu0oqh-Checkbox-icon-Checkbox-unchecked"
        font-size="medium"
      />
    </span>
    <span
      class="MuiTypography-root MuiTypography-body1 tss-5-Checkbox-label-ref tss-fxezff-Checkbox-label-MenuItem-label MuiFormControlLabel-label mui-b600p5-MuiTypography-root"
    >
      <span
        class="tss-0-Typography-weightRegular-ref tss-6tt2d9-Typography-root-Typography-weightRegular-Typography-body500"
      >
        <div
          class="tss-0-Typography-weightRegular-ref tss-10hdle-Typography-root-Typography-weightRegular-Tooltip-wrapper-TruncatedText-truncatedText-MenuItem-label-Typography-body400"
          data-mui-internal-clone-element="true"
        >
          <span
            class="tss-0-Typography-weightRegular-ref tss-1hhfpop-Typography-root-Typography-weightRegular-TruncatedText-truncatedText-Typography-body500"
          >
            Item Label
          </span>
        </div>
      </span>
    </span>
  </label>
</li>
`;

exports[`MenuItem Should render stories correctly story: "component": component 1`] = `
<li
  class="tss-1qfvypo-MenuItem-root MuiMenuItem-root MuiMenuItem-gutters MuiButtonBase-root  mui-t7a2rv-MuiButtonBase-root-MuiMenuItem-root"
  role="menuitem"
  tabindex="-1"
>
  <button
    class="MuiButtonBase-root tss-6-MenuItem-wrapper-ref tss-19jpft6-ButtonBase-root-ButtonBase-focusInset-MenuItem-wrapper mui-10d1a0h-MuiButtonBase-root"
    tabindex="0"
    type="button"
  >
    <div
      class="tss-sqs6xo-MenuItem-labelWrapper-MenuItem-label"
    >
      <div>
        Component Label
      </div>
    </div>
  </button>
</li>
`;

exports[`MenuItem Should render stories correctly story: "end icon": end icon 1`] = `
<li
  class="tss-1qfvypo-MenuItem-root MuiMenuItem-root MuiMenuItem-gutters MuiButtonBase-root  mui-t7a2rv-MuiButtonBase-root-MuiMenuItem-root"
  role="menuitem"
  tabindex="-1"
>
  <button
    class="MuiButtonBase-root tss-6-MenuItem-wrapper-ref tss-19jpft6-ButtonBase-root-ButtonBase-focusInset-MenuItem-wrapper mui-10d1a0h-MuiButtonBase-root"
    tabindex="0"
    type="button"
  >
    <div
      class="tss-sqs6xo-MenuItem-labelWrapper-MenuItem-label"
    >
      <div
        class="tss-0-Typography-weightRegular-ref tss-10hdle-Typography-root-Typography-weightRegular-Tooltip-wrapper-TruncatedText-truncatedText-MenuItem-label-Typography-body400"
        data-mui-internal-clone-element="true"
      >
        <span
          class="tss-0-Typography-weightRegular-ref tss-1hhfpop-Typography-root-Typography-weightRegular-TruncatedText-truncatedText-Typography-body500"
        >
          Item Label
        </span>
      </div>
    </div>
    <div
      class="tss-6-MenuItem-endIcon-ref tss-1q0x03j-Icon-root-Icon-sizeMd-MenuItem-endIcon"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium mui-i4bv87-MuiSvgIcon-root"
        color="inherit"
        focusable="false"
        viewBox="0 0 24 24"
      >
        angleRight.svg
      </svg>
    </div>
  </button>
</li>
`;

exports[`MenuItem Should render stories correctly story: "inline message": inline message 1`] = `
<li
  class="tss-1qfvypo-MenuItem-root MuiMenuItem-root MuiMenuItem-gutters MuiButtonBase-root  mui-t7a2rv-MuiButtonBase-root-MuiMenuItem-root"
  role="menuitem"
  tabindex="-1"
>
  <button
    class="MuiButtonBase-root tss-6-MenuItem-wrapper-ref tss-19jpft6-ButtonBase-root-ButtonBase-focusInset-MenuItem-wrapper mui-10d1a0h-MuiButtonBase-root"
    tabindex="0"
    type="button"
  >
    <div
      class="tss-sqs6xo-MenuItem-labelWrapper-MenuItem-label"
    >
      <div
        class="tss-0-Typography-weightRegular-ref tss-10hdle-Typography-root-Typography-weightRegular-Tooltip-wrapper-TruncatedText-truncatedText-MenuItem-label-Typography-body400"
        data-mui-internal-clone-element="true"
      >
        <span
          class="tss-0-Typography-weightRegular-ref tss-1hhfpop-Typography-root-Typography-weightRegular-TruncatedText-truncatedText-Typography-body500"
        >
          Item Label
        </span>
      </div>
    </div>
    <div
      class="tss-7-InlineMessage-sizeSm-ref tss-6-MenuItem-inlineMessage-ref tss-1cx58eo-InlineMessage-root-InlineMessage-sizeSm-InlineMessage-statusSuccess-MenuItem-inlineMessage"
    >
      <div
        class="tss-7-InlineMessage-icon-ref tss-3c0qi4-Icon-root-Icon-sizeSm-InlineMessage-icon"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium mui-i4bv87-MuiSvgIcon-root"
          color="inherit"
          focusable="false"
          viewBox="0 0 24 24"
        >
          checkCircle.svg
        </svg>
      </div>
      <div>
        <span
          class="tss-0-Typography-weightRegular-ref tss-1r4pp6w-Typography-root-Typography-weightRegular-InlineMessage-message-Typography-body400"
        >
          Success message
        </span>
      </div>
    </div>
  </button>
</li>
`;

exports[`MenuItem Should render stories correctly story: "start icon with sub label": start icon with sub label 1`] = `
<li
  class="tss-1qfvypo-MenuItem-root MuiMenuItem-root MuiMenuItem-gutters MuiButtonBase-root  mui-t7a2rv-MuiButtonBase-root-MuiMenuItem-root"
  role="menuitem"
  tabindex="-1"
>
  <button
    class="MuiButtonBase-root tss-6-MenuItem-wrapper-ref tss-19jpft6-ButtonBase-root-ButtonBase-focusInset-MenuItem-wrapper mui-10d1a0h-MuiButtonBase-root"
    tabindex="0"
    type="button"
  >
    <div
      class="tss-6-MenuItem-startIcon-ref tss-2hk6gu-Icon-root-Icon-sizeMd-MenuItem-startIcon"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium mui-i4bv87-MuiSvgIcon-root"
        color="inherit"
        focusable="false"
        viewBox="0 0 24 24"
      >
        fileDownload.svg
      </svg>
    </div>
    <div
      class="tss-sqs6xo-MenuItem-labelWrapper-MenuItem-label"
    >
      <div
        class="tss-0-Typography-weightRegular-ref tss-10hdle-Typography-root-Typography-weightRegular-Tooltip-wrapper-TruncatedText-truncatedText-MenuItem-label-Typography-body400"
        data-mui-internal-clone-element="true"
      >
        <span
          class="tss-0-Typography-weightRegular-ref tss-1hhfpop-Typography-root-Typography-weightRegular-TruncatedText-truncatedText-Typography-body500"
        >
          Item Label
        </span>
      </div>
      <div
        class="tss-0-Typography-weightRegular-ref tss-9b17g2-Typography-root-Typography-weightRegular-Tooltip-wrapper-TruncatedText-truncatedText-Typography-body400"
        data-mui-internal-clone-element="true"
      >
        <div
          class="tss-0-Typography-weightRegular-ref tss-6-MenuItem-subLabel-ref tss-3u963p-Typography-root-Typography-weightRegular-TruncatedText-truncatedText-MenuItem-subLabel-Typography-body400-Typography-secondary"
        >
          Extra info
        </div>
      </div>
    </div>
  </button>
</li>
`;

exports[`MenuItem Should render stories correctly story: "start icon": start icon 1`] = `
<li
  class="tss-1qfvypo-MenuItem-root MuiMenuItem-root MuiMenuItem-gutters MuiButtonBase-root  mui-t7a2rv-MuiButtonBase-root-MuiMenuItem-root"
  role="menuitem"
  tabindex="-1"
>
  <button
    class="MuiButtonBase-root tss-6-MenuItem-wrapper-ref tss-19jpft6-ButtonBase-root-ButtonBase-focusInset-MenuItem-wrapper mui-10d1a0h-MuiButtonBase-root"
    tabindex="0"
    type="button"
  >
    <div
      class="tss-6-MenuItem-startIcon-ref tss-2hk6gu-Icon-root-Icon-sizeMd-MenuItem-startIcon"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium mui-i4bv87-MuiSvgIcon-root"
        color="inherit"
        focusable="false"
        viewBox="0 0 24 24"
      >
        fileDownload.svg
      </svg>
    </div>
    <div
      class="tss-sqs6xo-MenuItem-labelWrapper-MenuItem-label"
    >
      <div
        class="tss-0-Typography-weightRegular-ref tss-10hdle-Typography-root-Typography-weightRegular-Tooltip-wrapper-TruncatedText-truncatedText-MenuItem-label-Typography-body400"
        data-mui-internal-clone-element="true"
      >
        <span
          class="tss-0-Typography-weightRegular-ref tss-1hhfpop-Typography-root-Typography-weightRegular-TruncatedText-truncatedText-Typography-body500"
        >
          Item Label
        </span>
      </div>
    </div>
  </button>
</li>
`;

exports[`MenuItem Should render stories correctly story: "sub label": sub label 1`] = `
<li
  class="tss-1qfvypo-MenuItem-root MuiMenuItem-root MuiMenuItem-gutters MuiButtonBase-root  mui-t7a2rv-MuiButtonBase-root-MuiMenuItem-root"
  role="menuitem"
  tabindex="-1"
>
  <button
    class="MuiButtonBase-root tss-6-MenuItem-wrapper-ref tss-19jpft6-ButtonBase-root-ButtonBase-focusInset-MenuItem-wrapper mui-10d1a0h-MuiButtonBase-root"
    tabindex="0"
    type="button"
  >
    <div
      class="tss-sqs6xo-MenuItem-labelWrapper-MenuItem-label"
    >
      <div
        class="tss-0-Typography-weightRegular-ref tss-10hdle-Typography-root-Typography-weightRegular-Tooltip-wrapper-TruncatedText-truncatedText-MenuItem-label-Typography-body400"
        data-mui-internal-clone-element="true"
      >
        <span
          class="tss-0-Typography-weightRegular-ref tss-1hhfpop-Typography-root-Typography-weightRegular-TruncatedText-truncatedText-Typography-body500"
        >
          Item Label
        </span>
      </div>
      <div
        class="tss-0-Typography-weightRegular-ref tss-9b17g2-Typography-root-Typography-weightRegular-Tooltip-wrapper-TruncatedText-truncatedText-Typography-body400"
        data-mui-internal-clone-element="true"
      >
        <div
          class="tss-0-Typography-weightRegular-ref tss-6-MenuItem-subLabel-ref tss-3u963p-Typography-root-Typography-weightRegular-TruncatedText-truncatedText-MenuItem-subLabel-Typography-body400-Typography-secondary"
        >
          Extra info
        </div>
      </div>
    </div>
  </button>
</li>
`;

exports[`MenuItem Should render stories correctly story: "tag": tag 1`] = `
<li
  class="tss-1qfvypo-MenuItem-root MuiMenuItem-root MuiMenuItem-gutters MuiButtonBase-root  mui-t7a2rv-MuiButtonBase-root-MuiMenuItem-root"
  role="menuitem"
  tabindex="-1"
>
  <button
    class="MuiButtonBase-root tss-6-MenuItem-wrapper-ref tss-19jpft6-ButtonBase-root-ButtonBase-focusInset-MenuItem-wrapper mui-10d1a0h-MuiButtonBase-root"
    tabindex="0"
    type="button"
  >
    <div
      class="tss-sqs6xo-MenuItem-labelWrapper-MenuItem-label"
    >
      <div
        class="tss-0-Typography-weightRegular-ref tss-10hdle-Typography-root-Typography-weightRegular-Tooltip-wrapper-TruncatedText-truncatedText-MenuItem-label-Typography-body400"
        data-mui-internal-clone-element="true"
      >
        <span
          class="tss-0-Typography-weightRegular-ref tss-1hhfpop-Typography-root-Typography-weightRegular-TruncatedText-truncatedText-Typography-body500"
        >
          Item Label
        </span>
      </div>
    </div>
    <div
      class="tss-6-MenuItem-tag-ref tss-4u6r2-Tag-root-Tag-colorPrimary-MenuItem-tag"
    >
      <span
        class="tss-0-Typography-weightRegular-ref tss-j8p12z-Typography-root-Typography-weightRegular-Typography-body400"
      >
        123
      </span>
    </div>
  </button>
</li>
`;

exports[`MenuItem Should render stories correctly story: "text": text 1`] = `
<li
  class="tss-1qfvypo-MenuItem-root MuiMenuItem-root MuiMenuItem-gutters MuiButtonBase-root  mui-t7a2rv-MuiButtonBase-root-MuiMenuItem-root"
  role="menuitem"
  tabindex="-1"
>
  <button
    class="MuiButtonBase-root tss-6-MenuItem-wrapper-ref tss-19jpft6-ButtonBase-root-ButtonBase-focusInset-MenuItem-wrapper mui-10d1a0h-MuiButtonBase-root"
    tabindex="0"
    type="button"
  >
    <div
      class="tss-sqs6xo-MenuItem-labelWrapper-MenuItem-label"
    >
      <div
        class="tss-0-Typography-weightRegular-ref tss-10hdle-Typography-root-Typography-weightRegular-Tooltip-wrapper-TruncatedText-truncatedText-MenuItem-label-Typography-body400"
        data-mui-internal-clone-element="true"
      >
        <span
          class="tss-0-Typography-weightRegular-ref tss-1hhfpop-Typography-root-Typography-weightRegular-TruncatedText-truncatedText-Typography-body500"
        >
          Item Label
        </span>
      </div>
    </div>
  </button>
</li>
`;
